ros_buildfarm/argument.py

```diff
@@ -281,7 +281,7 @@ def add_argument_package_names(parser):
 def add_argument_repository_names(parser):
     parser.add_argument(
         '--repository-names',
-        nargs='+',
+        nargs='*',
```

Comment:
I noticed you added some `assert`s to guard against specifying neither `--repository-names` or `--repos-file-urls`. Since specifying `+` previously guarded against an empty package set, and there are consumers of `add_argument_repository_names` which don't also use `add_argument_repos_file_urls`, shouldn't the existing behavior be maintained?

Maybe using an optional argument `required=True` which determines if `+` or `*` is used?

Reply:
Good point: 381dec08fe58a0456f07629e3678a45df9739309 keeps the previous semantic of `nargs='+'` by default and all new usages explicitly choose to make it optional.